### DISCUSSION TITLE
Make Overhangs printable

### DISCUSSION THREAD

**Discussion by Skyl-Orca** [OrcaSlicer/OrcaSlicer#1319] [Ideas]
I've been using the OrcaSlicer for a while now and I'm super happy with it, especially in combination with the clipper it's an absolute dream. Unfortunately, I'm still missing a setting that I used with Cura often and with pleasure: the Make Overhang printable option.
The option is added under an overhang "Material added". This is especially helpful for decorative things that don't necessarily have to be a perfect fit, since you save on supporting structures and the models are printed cleanly. Could this option be included in later versions? :-)

**Comment by odie5533** [comment]
Cura has this feature in their CuraEngine. They configure the setting as conical_overhang_enabled (which they show in the interface as "Make Overhangs Printable") https://github.com/Ultimaker/CuraEngine/blob/main/src/FffPolygonGenerator.cpp#L233

And the actual ConicalOverhang function is in https://github.com/Ultimaker/CuraEngine/blob/main/src/ConicalOverhang.cpp

![Make overhangs printable](https://github.com/SoftFever/OrcaSlicer/assets/371066/dff97ea9-d14e-43b4-8264-b622716d9718)

**Comment by odie5533** [comment]
I would offer a $300 Bounty for this feature.

**Comment by SoftFever** [comment]
I personally never used this feature as I have concerns that the geometry changes is uncontrollable.
Can you list some use cases? It would be perfect if you can also provide some sample models(real life model)

**Comment by odie5533** [comment]
Thank you for considering this feature!

The setting must be used judiciously, but it saves having to use supports or modify the model. It has a wide range of uses from organics like statues, functional prints, hard shapes, and soft shapes. As long as the changes are subtle enough or in unimportant places, it can be a godsend since it avoids supports, saving print time and cleanup.

Cura includes two adjustment settings: Maximum Model Angle (conical_overhang_angle) and Maximum Overhang Hole Area (conical_overhang_hole_size).

[knob.stl.zip](https://github.com/SoftFever/OrcaSlicer/files/11906500/knob.stl.zip) from [thingiverse](https://www.thingiverse.com/thing:665410/files)

![ezgif com-optimize(1)](https://github.com/SoftFever/OrcaSlicer/assets/371066/159bc73b-ea04-4e3c-87a5-d22632685e8a)

**Comment by odie5533** [comment]
This example modifies under the ears and under the chin.

https://www.thingiverse.com/thing:276994/files

![ezgif com-optimize](https://github.com/SoftFever/OrcaSlicer/assets/371066/7be1c2dc-2921-44bd-a031-56a2d86bf90c)

![ezgif com-optimize(1)](https://github.com/SoftFever/OrcaSlicer/assets/371066/911eed54-78d6-4e5b-9b1f-ef70824ad76a)

**Reply by SoftFever** [reply]
Thanks for the images.
I will evaluate this when I got time.

**Comment by odie5533** [comment]
I made an attempt. I did not get very far. Added a setting and started on the algorithm, but I am unable to progress. Hopefully someone else may be encouraged to make an attempt!

https://github.com/odie5533/OrcaSlicer/pull/1/files

**Comment by odie5533** [comment]
I've created Issue #1534 which includes some demo stl files (a block with an overhang) to test on and more pictures.

I've increased the bounty to $300.

**Comment by Tsukotaku** [comment]
I follow up for this feature request, it's the only thing that's keeping me tide to CURA right now.

**Comment by odie5533** [comment]
@Skyl-Orca @Tsukotaku Thanks to Noisyfox's work, a dev PR is up!

Windows build: https://github.com/SoftFever/OrcaSlicer/actions/runs/5581868427?pr=1556
